core/tests/protractor_utils/users.js

```diff
@@ -17,85 +17,51 @@
-    email, isSuperAdmin = false, manualNavigation = true) {
-  // Use of element and action is not possible because the login page
-  // is non-angular.
-  // The full url is also necessary.
-  var driver = browser.driver;
-  // The manualNavigation argument is used to determine whether to navigate to
-  // the login URL using driver.get() or not. If false, the calling method
-  // should handle navigation to the login page.
-  if (manualNavigation) {
-    await driver.get(general.SERVER_URL_PREFIX + general.LOGIN_URL_SUFFIX);
-  }
-  // The statement below uses a browser.wait() to determine if the user has
-  // logged in. Use of waitFor is not possible because the active page is
-  // non-angular.
-  await browser.wait(
-    async() => {
-      try {
-        await driver.findElement(protractor.By.name('email'));
-      } catch (error) {
-        return false;
-      }
-      return true;
-    }, waitFor.DEFAULT_WAIT_TIME_MSECS, 'Login takes too long.');
-  await (await driver.findElement(protractor.By.name('email'))).clear();
-  await (await driver.findElement(protractor.By.name('email'))).sendKeys(email);
+var _createFirebaseAccount = async function(email, isSuperAdmin = false) {
+  var user = await FirebaseAdmin.auth().createUser({
+    email: email,
+    emailVerified: true,
+    password: await HashWasm.md5(email),
+  });
   if (isSuperAdmin) {
-    await (await driver.findElement(protractor.By.name('admin'))).click();
-    let adminCheckboxStatus = await driver.findElement(
-      protractor.By.name('admin')).getAttribute('checked');
-    expect(adminCheckboxStatus).toBeTruthy();
-  }
-  await (await driver.findElement(protractor.By.id('submit-login'))).click();
-  if (manualNavigation) {
-    // The statement below uses a browser.wait() to determine if the user has
-    // logged in. Use of waitFor is not possible because the active page is
-    // non-angular.
-    await browser.wait(
-      async() => {
-        let loginStatusHeaderElement = (
-          await driver.findElement(protractor.By.tagName('h3')));
-        let text = await loginStatusHeaderElement.getText();
-        return text !== 'Logged In';
-      }, waitFor.DEFAULT_WAIT_TIME_MSECS, 'Login takes too long.');
+    await FirebaseAdmin.auth().setCustomUserClaims(
+      user.uid, {role: 'super_admin'});
   }
 };
 
-var login = async function(email, manualNavigation = true) {
-  await _login(email, false, manualNavigation);
+var login = async function(email) {
+  // Use of element and action is not possible because the login page is
+  // non-angular. The full url is also necessary.
+  await browser.driver.get(
+    general.SERVER_URL_PREFIX + general.LOGIN_URL_SUFFIX);
+
+  await waitFor.alertToBePresent();
+  const alert = await browser.switchTo().alert();
+  await alert.sendKeys(email);
+  await alert.accept();
+
+  await waitFor.pageToFullyLoad();
 };
 
 var logout = async function() {
-  // Use of action is not possible because logout page is non-angular.
```

Comment:
Maybe keep the comment :)

Reply:
I think the login and logout pages are now in Angular.

Reply:
The comment on lines 41 and 42 above says the login page is still not in Angular:

```js
  // Use of element and action is not possible because the login page is
  // non-angular. The full url is also necessary.
```

However, it looks like we're just logging out from the profile menu, so is there even really a "logout" page?

Reply:
We don't always "start" from an Angular page. Have updated comments to make this clearer.